packages/playwright-core/src/server/frames.ts

```diff
@@ -1372,6 +1372,14 @@ export class Frame extends SdkObject {
   }
 
   async expect(metadata: CallMetadata, selector: string, options: FrameExpectParams): Promise<{ matches: boolean, received?: any, log?: string[], timedOut?: boolean }> {
+    const result = await this._expectImpl(metadata, selector, options);
+    // Library mode special case for the expect errors which are return values, not exceptions.
+    if (result.matches === options.isNot)
+      metadata.fakeError = { error: { name: 'Expect', message: 'Expect failed' }};
```

Comment:
Lets use a symbol instead?
We still send fakeError to the client, would love to never send it to the client.